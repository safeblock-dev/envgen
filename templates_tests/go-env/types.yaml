options:
  package: types

types:
  - name: Environment
    type: string
    description: Application environment
    values: [development, staging, production]
  - name: URL
    type: "*url.URL"
    description: URL type
    import: "net/url"
  - name: Duration
    type: "time.Duration"
    description: Duration type
    import: "time"
  - name: IpAddresses
    type: "[]net.IP"
    description: IP addresses type
    import: "net"

groups:
  - name: App
    description: Application settings
    fields:
      - name: env
        type: Environment
        description: Application environment
        required: true
        example: "development"
      - name: api_url
        type: URL
        description: API endpoint
        required: true
        example: "https://api.example.com"
      - name: request_timeout
        type: Duration
        description: API request timeout
        default: "30s"
      - name: response_timeout
        type: Duration
        description: API response timeout
        default: "30s"
      - name: allowed_ips
        type: "IpAddresses"
        description: List of allowed IP addresses
        example: "127.0.0.1,192.168.1.1" 