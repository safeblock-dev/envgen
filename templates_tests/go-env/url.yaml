options:
  go_meta: |
    // is URL? {{ isURL getTemplatePath }}

groups:
  - name: App
    description: Basic application settings
    fields:
      - name: Debug
        type: bool
        description: Enable debug mode
        default: "false"
      - name: Port
        type: int
        description: Server port
        required: true 