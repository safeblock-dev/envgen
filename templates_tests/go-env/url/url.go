// Code generated by envgen. DO NOT EDIT.
// This file was automatically generated and should not be modified manually.

//go:generate envgen gen -c ../url.yaml -o url.generated -t https://raw.githubusercontent.com/safeblock-dev/envgen/main/templates/go-env

package url

// AppConfig represents Basic application settings
type AppConfig struct {
	Debug bool `env:"DEBUG" envDefault:"false"` // Enable debug mode
	Port int `env:"PORT,required"` // Server port
}