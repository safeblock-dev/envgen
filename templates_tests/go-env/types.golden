// Code generated by envgen. DO NOT EDIT.
// This file was automatically generated and should not be modified manually.

//go:generate go run github.com/safeblock-dev/envgen/cmd/envgen -c types.yaml -o types.generated -t ../../templates/go-env

package types
import (
	"net/url"
	"time"
	"net"
)

// AppConfig represents Application settings
type AppConfig struct {
	Env Environment `env:"ENV,required"` // Application environment (Possible values: development, staging, production)
	ApiUrl *url.URL `env:"API_URL,required"` // API endpoint URL
	Timeout time.Duration `env:"TIMEOUT" envDefault:"30s"` // API request timeout
	AllowedIps []net.IP `env:"ALLOWED_IPS"` // List of allowed IP addresses
}